### DISCUSSION TITLE
How to use hyper to send requests and receive responses byte by byte?

### DISCUSSION THREAD

**Discussion by Anivie** [hyperium/hyper#3552] [Q&A]
Hello community,
I'm sorry about just starting to use this library, so I'm not familiar with its API.
My goal is to start processing the data in the body immediately when the server returns it.
This means that I need to start processing immediately while receiving is still ongoing, instead of waiting for the reception to be completed before starting.
Currently, I have found two APIs: `hyper_util::client::legacy::client` and `hyper::client::conn::http2::handshake`. However, the former cannot process data immediately while receiving (`request.await` needs to block until all data is received), and for the latter, I haven't found how to use it in HTTPS or if there are similar APIs available. Does anyone know how should I handle this?

**Comment by seanmonstar** [comment]
>  the former cannot process data immediately while receiving (request.await needs to block until all data is received)

That's not the case. hyper streams bodies. You await for the receipt of the response headers, and then you can await chunks of the body data as they arrive. You can see an example of that in [client basic guide](https://hyper.rs/guides/1/client/basic/).
